File: server-rails/app/lib/clients/twitter_client.rb
require 'twitter'

class Clients::TwitterClient
  @client = Twitter::REST::Client.new do |config|
    config.consumer_key = AppEnv['TWITTER_CONSUMER_KEY']
    config.consumer_secret = AppEnv['TWITTER_CONSUMER_SECRET']
    config.access_token = AppEnv['TWITTER_ACCESS_TOKEN']
    config.access_token_secret = AppEnv['TWITTER_ACCESS_SECRET']
  end

  def self.latest_tweets
    # API gets count amt, then filters out rts and replies
    # This guarantees there will always be a tweet
    feed = @client
           .user_timeline('mojotech', count: 5, include_rts: true,
                                      exclude_replies: false, tweet_mode: 'extended')
    feed.map { |t| Clients::CreateTweet.new(t) }
  end
end

File: server-phoenix/lib/helios_web/clients/twitter_client.ex
defmodule HeliosWeb.Clients.TwitterClient do
  alias HeliosWeb.Clients.Tweet

  def latest_tweets do
    feed =
      ExTwitter.user_timeline(
        screen_name: "MojoTech",
        count: 5,
        include_rts: true,
        exclude_replies: false,
        tweet_mode: "extended"
      )

    Enum.map(feed, fn tweet -> Tweet.from_api(tweet) end)
  end
end
